askama_shared/src/lib.rs

```diff
@@ -19,6 +19,47 @@ pub use generator::generate;
 mod generator;
 mod parser;
 
+use std::fmt::{self, Display, Formatter};
+
+pub enum MarkupDisplay<'a, T> where T: 'a + Display {
+    Safe(&'a T),
```

Comment:
I think you don't need to store a reference to `T`, you can just make it `T`. In that case (`T: Display`) it can store both `T` and `&T` (as well as `&mut T`), because `Display` [is implemented for references to `T`](https://play.rust-lang.org/?gist=09f16050ae06f899b588bb59db6a8ab8&version=stable).